PyInstaller/hooks/hook-eyed3.plugins.py

```diff
@@ -0,0 +1,12 @@
+from PyInstaller.utils.hooks import collect_data_files, collect_submodules
+from os.path import basename, splitext
+
+plugins = filter(lambda data: basename(data[0])[0] not in ('_', '.') and data[0].endswith('.py'),
+        collect_data_files('eyed3.plugins', include_py_files=True))
+
+datas = plugins
+
+imports = list(set(map(lambda data: splitext(basename(data[0]))[0],
+        datas)))
+
+hiddenimports = imports
```

Comment:
This looks like being wrong. Either add the plugins as submodules/hiddenimport *or* as data-files. Why is there a need to do both? Please recheck.

Reply:
I found that PyInstaller cannot import data-files, or maybe I'm wrong.
It just worked when I add them both.
Is there any recipes?

Reply:
Well, as these are data-file, they are not intended to be imported. You could tkeak your application to do so, but this is less useful as it requires every application developer to to the same.

There is no reciept to work around such a resticted implementation.

But again please use more readable code like this:
```
# TODO: remove `.py` extension
hiddenimports = ["eye3d.plugins." + os.path.basename(data[0])
    for data in datas]
```
No need to filter duplicates,